app/controllers/album.js

```diff
@@ -1,7 +1,62 @@
-const albumService = require('../services/album');
+const albumService = require('../services/album'),
+  albumInteractor = require('../interactors/album'),
+  error = require('../errors'),
+  logger = require('../logger'),
+  config = require('../../config'),
+  { validationResult } = require('express-validator/check');
 
 exports.list = (req, res, next) => {
 
-  albumService.listRequest(req, res, next);
+  logger.info(`Attempting GET request to url ${config.common.urlRequests.albumList}`);
+
+  albumService.listRequest().then(albums => {
+    logger.info(`Response received from ${config.common.urlRequests.albumList}`);
+    return res.status(200).send(JSON.parse(albums));
+  }).catch(error => {
+    logger.error(`An error occured while attempting a GET request to url ${config.common.urlRequests.albumList}: ${error.message}`);
+    return res.status(502).send(error.message);
+  });
+
+};
+
+exports.purchase = (req, res, next) => {
+
+  try {
+    validationResult(req).throw();
+  } catch (err) {
+    return next(error.invalidAlbumId);
+  }
+
+  logger.info(`User ${req.user.email} requested to purchase an Album, ID ${req.body.id}`);
+
+  albumService.checkAlbum(req.body.id).then(album => {
+    if(!album){
+      return next(errors.notAnAlbum);
+    }else{
```

Comment:
The advantage of using the above return is that you can avoid the else statement.